src/runtime/virtcontainers/utils/utils_linux.go

```diff
@@ -151,3 +153,59 @@ func IsAPVFIOMediatedDevice(sysfsdev string) bool {
 	}
 	return false
 }
+
+func waitProcessUsingPidfd(pid int, timeoutSecs uint, logger *logrus.Entry) (bool, error) {
+	pidfd, err := unix.PidfdOpen(pid, 0)
+
+	if err != nil {
+		if err == unix.ESRCH {
+			return false, nil
+		}
+
+		return true, err
+	}
+
+	defer unix.Close(pidfd)
+
+	start := time.Now()
+	var n int
+
+	for {
+		remaining := int(timeoutSecs*1000) - int(time.Since(start).Milliseconds())
```

Comment:
It's a bit disconcerting to mix `time.Duration` and times in milliseconds represented as `int`

What about something like:

```
maxDelay := time.Duration(timeoutSecs * time.Second)
end := time.Now().Add(maxDelay)
for {
   remaining := time.Until(end).Milliseconds()
[...]
   Poll(..., int(remaining))
```

I would find this more readable than multiplications by 1000 and similar.

Reply:
Fixed.